src/schema.ts

```diff
@@ -172,13 +172,28 @@ export class Schema {
           case TimeUnit.MILLISECONDS: return 1000;
         }
       }
+      var unit = encQ.timeUnit as string;
+      var timeStats = fieldSchema.timeStats;
       // if the cardinality for the timeUnit is not cached, calculate it
-      if (!fieldSchema.timeStats[encQ.timeUnit as string]) {
-        fieldSchema.timeStats[encQ.timeUnit as string] = timeSummary(encQ.timeUnit as TimeUnit, fieldSchema.stats);
+      if (!timeStats[unit]) {
+        timeStats[unit] = timeSummary(encQ.timeUnit as TimeUnit, fieldSchema.stats);
+      }
+
+      if (excludeInvalid) {
+        return timeStats[unit].distinct - invalidCount(timeStats[unit].unique, ['Invalid Date', null]);
+      } else {
+        return timeStats[unit].distinct;
       }
-      return fieldSchema.timeStats[encQ.timeUnit as string].distinct;
     } else {
-      return fieldSchema ? fieldSchema.stats.distinct : null;
+      if (fieldSchema) {
+        if (excludeInvalid) {
+          return fieldSchema.stats.distinct - invalidCount(fieldSchema.stats.unique, ['null', 'NaN']);
```

Comment:
Why is this one `'null'` and the above is `null`.  I guess it doesn't matter but they should be consistent.